pkg/ddl/ingest/checkpoint.go

```diff
@@ -438,3 +480,35 @@ func (s *CheckpointManager) updateCheckpointLoop() {
 		}
 	}
 }
+
+func (s *CheckpointManager) updateCheckpoint() error {
+	failpoint.Inject("checkpointLoopExit", func() {
+		// used in a manual test
+		failpoint.Return(errors.New("failpoint triggered so can't update checkpoint"))
+	})
+	finishCh := make(chan struct{})
+	select {
+	case s.updaterCh <- finishCh:
+	case <-s.ctx.Done():
+		return s.ctx.Err()
+	}
+	// wait updateCheckpointLoop to finish checkpoint update.
+	select {
+	case <-finishCh:
+	case <-s.ctx.Done():
+		return s.ctx.Err()
+	}
+	return nil
+}
+
+func (s *CheckpointManager) refreshTS() (uint64, error) {
```

Comment:
```suggestion
func (s *CheckpointManager) refreshTSAndUpdateCP() (uint64, error) {
```
the refreshTS part is same as in `resumeOrInitCheckpoint`